include/micm/solver/lu_decomposition.inl

```diff
@@ -223,16 +213,19 @@ namespace micm
             L_vector[lki_nkj->first] -= L_vector[lkj_uji->first] * U_vector[lkj_uji->second];
             ++lkj_uji;
           }
+
           if (U_vector[*uii] == 0.0)
           {
             is_singular = true;
-            return;
```

Comment:
Shall we still `return` here if we already know it is singular?

Reply:
that only makes sense when we want to check for singularity. We haven't passed that information down here so we have to continue and expect the rosenbrock implementation to handle singularity properly

Reply:
Ah, that makes sense. Thanks for the clarification.